topi/python/topi/tensor_overload_op.py

```diff
@@ -0,0 +1,113 @@
+# pylint: disable=invalid-name,unused-argument,protected-access
+"""Overloaded binary operators"""
+from __future__ import absolute_import as _abs
+import tvm
+from . import broadcast as _broadcast
+
+
+def _tensor_broadcast_bop(lhs, rhs, op):
```

Comment:
use functional programming style to build a generic functor that returns functions.

So we can do things like tvm.generic.add = _make_bop_impl(lambda x, y: x+y, _broadcast.broadcast_add)